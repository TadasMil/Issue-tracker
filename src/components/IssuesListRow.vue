<script setup lang="ts" name="IssuesListRow">
import { ISSUE_STATUS } from '@/constnats'
import type { IIssueStatus } from '@/types'
import type { ColorVariant } from 'bootstrap-vue-3'

import { computed } from 'vue'

const props = defineProps<{
  title: string
  description: string
  status?: IIssueStatus
  trashed?: boolean
}>()

type colorVariantType = {
  [key in IIssueStatus]: ColorVariant
}

const statusLookup: colorVariantType = {
  [ISSUE_STATUS.COMPLETED]: 'success',
  [ISSUE_STATUS.OPEN]: 'warning',
}

const statusColor = computed(
  () => statusLookup[props.status || ISSUE_STATUS.OPEN]
)
</script>

<template>
  <b-card class="shadow-sm border-0 mb-3">
    <div class="d-flex justify-content-between">
      <b-container fluid>
        <b-row>
          <b-col v-if="trashed" cols="4" lg="2">
            <b-card-text>
              <p class="mb-1">Status</p>

              <div>
                <b-badge pill :variant="statusColor" class="text-uppercase">
                  {{ status }}
                </b-badge>
              </div>
            </b-card-text>
          </b-col>

          <b-col cols="4" lg="2">
            <b-card-text>
              <p class="mb-1">Title:</p>
              <h5 class="text-truncate mr-2">{{ title }}</h5>
            </b-card-text>
          </b-col>
          <b-col cols="4" lg="6">
            <b-card-text>
              <p class="mb-1">Description:</p>
              <h5 class="text-truncate mr-2">{{ description }}</h5>
            </b-card-text>
          </b-col>
        </b-row>
      </b-container>

      <slot />
    </div>
  </b-card>
</template>
