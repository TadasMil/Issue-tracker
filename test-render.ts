import type { RenderOptions } from '@testing-library/vue'
import BootstrapVue3 from 'bootstrap-vue-3'
import { render } from '@testing-library/vue'

const customRender = (ui: any, options?: RenderOptions) =>
  render(ui, {
    ...options,
    global: {
      ...options?.global,
      plugins: [BootstrapVue3, ...(options?.global?.plugins || [])],
    },
  })

export * from '@testing-library/vue'
export { customRender as render }
