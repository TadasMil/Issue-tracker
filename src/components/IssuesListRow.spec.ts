import { describe, it, expect } from 'vitest'
import { render } from '../../test-render'

import IssuesListRow from './IssuesListRow.vue'

describe('IssuesListRow', () => {
  it('should render the component', () => {
    const { html } = render(IssuesListRow, {
      props: {
        title: 'Test title',
        description: 'Test description',
      },
      global: {
        stubs: ['ActionableDropdown'],
      },
    })

    expect(html()).toMatchSnapshot()
  })
})
