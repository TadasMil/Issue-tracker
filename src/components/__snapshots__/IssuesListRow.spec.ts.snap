// Vitest Snapshot v1

exports[`IssuesListRow > should render the component 1`] = `
"<div class=\\"card shadow-sm border-0 mb-3\\">
  <!---->
  <!---->
  <div class=\\"card-body\\">
    <!---->
    <!---->
    <div class=\\"d-flex justify-content-between\\">
      <div class=\\"container-fluid\\">
        <div class=\\"row\\">
          <!--v-if-->
          <div class=\\"col-lg-2 col-4\\">
            <p class=\\"card-text\\">
            <p class=\\"mb-1\\">Title:</p>
            <h5 class=\\"text-truncate mr-2\\">Test title</h5>
            </p>
          </div>
          <div class=\\"col-lg-6 col-4\\">
            <p class=\\"card-text\\">
            <p class=\\"mb-1\\">Description:</p>
            <h5 class=\\"text-truncate mr-2\\">Test description</h5>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <!---->
  <!---->
</div>"
`;
